form/src/main/java/org/dhis2/form/ui/provider/inputfield/CoordinatesProvider.kt

```diff
@@ -0,0 +1,76 @@
+package org.dhis2.form.ui.provider.inputfield
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import org.dhis2.commons.resources.ResourceManager
+import org.dhis2.form.R
+import org.dhis2.form.extensions.inputState
+import org.dhis2.form.extensions.legend
+import org.dhis2.form.extensions.supportingText
+import org.dhis2.form.model.FieldUiModel
+import org.dhis2.form.model.UiRenderType
+import org.dhis2.form.ui.event.RecyclerViewUiEvents
+import org.dhis2.form.ui.intent.FormIntent
+import org.hisp.dhis.android.core.arch.helpers.GeometryHelper
+import org.hisp.dhis.android.core.common.FeatureType
+import org.hisp.dhis.android.core.common.Geometry
+import org.hisp.dhis.mobile.ui.designsystem.component.Coordinates
+import org.hisp.dhis.mobile.ui.designsystem.component.InputCoordinate
+
+@Composable
+fun ProvideInputCoordinate(
+    modifier: Modifier,
+    fieldUiModel: FieldUiModel,
+    intentHandler: (FormIntent) -> Unit,
+    uiEventHandler: (RecyclerViewUiEvents) -> Unit,
+    resources: ResourceManager,
+) {
+    if (fieldUiModel.renderingType == UiRenderType.POINT) {
+        InputCoordinate(
+            title = fieldUiModel.label,
+            state = fieldUiModel.inputState(),
+            supportingText = fieldUiModel.supportingText(),
+            legendData = fieldUiModel.legend(),
+            coordinates = mapGeometry(fieldUiModel.value, FeatureType.POINT),
+            latitudeText = resources.getString(R.string.latitude),
+            longitudeText = resources.getString(R.string.longitude),
+            addLocationBtnText = resources.getString(R.string.add_location),
+            isRequired = fieldUiModel.mandatory,
+            modifier = modifier,
+            onResetButtonClicked = {
+                intentHandler.invoke(
+                    FormIntent.OnSave(
+                        uid = fieldUiModel.uid,
+                        value = null,
+                        valueType = fieldUiModel.valueType,
+                    ),
+                )
+            },
+            onUpdateButtonClicked = {
+                uiEventHandler.invoke(
+                    RecyclerViewUiEvents.RequestLocationByMap(
+                        uid = fieldUiModel.uid,
+                        featureType = FeatureType.POINT,
+                        value = fieldUiModel.value,
+                    ),
+                )
+            },
+        )
+    } else {
+        // TODO(ImplementPOlygon)
```

Comment:
There are two checks of rendertype point or polygon, here and in FieldProvider, we should only have one

Reply:
solved